crates/api/tests/stack-overflow.rs

```diff
@@ -0,0 +1,65 @@
+use std::sync::atomic::{AtomicUsize, Ordering::SeqCst};
+use wasmtime::*;
+
+#[test]
+fn host_always_has_some_stack() -> anyhow::Result<()> {
+    static HITS: AtomicUsize = AtomicUsize::new(0);
+    // assume hosts always have at least 512k of stack
+    const HOST_STACK: usize = 512 * 1024;
+
+    let store = Store::default();
+
+    // Create a module that's infinitely recursive, but calls the host on each
+    // level of wasm stack to always test how much host stack we have left.
+    let module = Module::new(
+        &store,
+        r#"
+            (module
+                (import "" "" (func $host))
+                (func $recursive (export "foo")
+                    call $host
+                    call $recursive)
+            )
+        "#,
+    )?;
+    let func = Func::wrap(&store, test_host_stack);
+    let instance = Instance::new(&module, &[func.into()])?;
+    let foo = instance
+        .get_export("foo")
+        .unwrap()
+        .func()
+        .unwrap()
```

Comment:
`func()` has been renamed to `into_func()`, but this can be simplified to just `instance.get_func("foo")`.